src/reflect/scala/reflect/internal/Symbols.scala

```diff
@@ -725,7 +725,7 @@ trait Symbols extends api.Symbols { self: SymbolTable =>
 
     def setFlag(mask: Long): this.type   = { _rawflags |= mask ; this }
     def resetFlag(mask: Long): this.type = { _rawflags &= ~mask ; this }
-    def resetFlags() { rawflags &= TopLevelCreationFlags }
+    def resetFlags() { rawflags = 0 }
```

Comment:
Could you add some motivation for the new behavior? It does seem pretty weird to preserve these flags... I could see something going between final and not-final, for example. Maybe we could strip the `TopLevelCreationFlags` to its essence?